### DISCUSSION TITLE
Health checks disable

### DISCUSSION THREAD

**Discussion by fate8383** [nextcloud/all-in-one#1617] [Questions] [answered]
Hi, is there anyway I can disable container health checks? I know how to do that with master container it works but all other containers created by master container have health checks thus the logs building up my syslog. I could not figure it out how.

**Comment by Zoey2936** [comment]
This is not possible, the only option would be to use [manual-install](https://github.com/nextcloud/all-in-one/tree/main/manual-install).
We could only redirect the output of the healthcheck to /dev/null. What do you think @szaimen?

**Comment by szaimen** [comment]
I wonder which container exactly is producing the logs? We are using nc for almost all healthcheck and that should not produce any logs, no?

**Reply by Zoey2936** [reply]
- Apache uses curl for port 8000
- clamav reports something like "working" as output
- Elasticsearch uses curl on port 9200
- Onlyoffice uses curl on port 80
- postgres also has an output
- redis hast "PONG" as output

**Reply by szaimen** [reply]
and where are these written to? Is there a healthcheck log somewhere?

**Reply by Zoey2936** [reply]
I can view the latest output using portainer, but it seems as they get also written into a syslog or something like this, as the authors says, but I don't know if and where it is written to exactly

**Reply by szaimen** [reply]
all right, then I would be grateful over a PR! However for clamav it would probably be recommended to create the pr upstream?

**Reply by Zoey2936** [reply]
> all right, then I would be grateful over a PR! However for clamav it would probably be recommended to create the pr upstream?

I think they wont accept it, as the output is wanted by their health check script (clamav)

**Reply by Zoey2936** [reply]
See: https://github.com/Cisco-Talos/clamav-docker/blob/fdc41e9d0a3833e39e0c4a2fb3d56da0c01ebdb2/clamav/0.105/alpine/scripts/clamdcheck.sh#L11

**Reply by fate8383** [reply]
Thanks for looking at it, yeah it writes to portioner and also syslog of the host continuously. Just one of them:

systemd[1]: run-docker-runtime\x2drunc-moby-2325a63a7a727530991644953eb7bc652929125bb6caee2a2fcef68cded69064-runc.0jLu8Q.mount: Succeeded.

**Reply by Zoey2936** [reply]
> Thanks for looking at it, yeah it writes to portioner and also syslog of the host continuously. Just one of them:
> 
> systemd[1]: run-docker-runtime\x2drunc-moby-2325a63a7a727530991644953eb7bc652929125bb6caee2a2fcef68cded69064-runc.0jLu8Q.mount: Succeeded.

Oh... I think this can not be resolved by #1619 that can be only done by completely removing the healthchecks

I think this should be reported to the moby GitHub repo, that docker spams systemd when running healthchecks
Because I see there more good points on healthchecks I wouldn't remove them from AIO, so you have three option
1. accept the logs
2. Maybe ask the moby (docker) projekt for an option to disable this logs
3. use manual install and disable healthchecks using a compose.override.yml file (but then many feauteres will get lost)

**Reply by fate8383** [reply]
healthcheck:
       disable: true

Simple entry to docker compose for master container stops the health check and stops the logs, I just want to be able to do with child containers which master auto creates. Is it possible?

**Reply by Zoey2936** [reply]
only if you use manual install

**Reply by fate8383** [reply]
Ok understood, thanks

**Reply by Zoey2936** [reply]
You could have a look at:
- https://github.com/docker/for-linux/issues/679#issuecomment-1004955357
- https://github.com/docker/for-linux/issues/679#issuecomment-509159724
- https://github.com/docker/for-linux/issues/679 in general

**Reply by Zoey2936** [reply]
If those work, you can continue using the normal installation including its advantages

**Comment by fate8383** [comment, marked as answer]
fantastic!
I created a new file /etc/rsyslog.d/01-blocklist.conf, containing:

if $msg contains "run-docker-runtime" and $msg contains ".mount: Succeeded." then {
    stop
}


this worked, the other option in systemd with conf didn't work. Thanks a lot!
